crates/tabby/src/routes/completions.rs

```diff
@@ -36,11 +36,38 @@ pub async fn completions(
 
     let user_agent = user_agent.map(|x| x.0.to_string());
 
+    let mut use_crlf = false;
+    if let Some(segments) = request.segments {
+        let mut new_segments = segments.clone();
+        if segments.prefix.contains("\r\n") {
+            use_crlf = true;
+            new_segments.prefix = segments.prefix.replace("\r\n", "\n");
+        }
+        if let Some(suffix) = segments.suffix {
+            if suffix.contains("\r\n") {
+                use_crlf = true;
+                new_segments.suffix = Some(suffix.replace("\r\n", "\n"));
+            }
+        }
+        request.segments = Some(new_segments);
+    }
+
     match state
         .generate(&request, &allowed_code_repository, user_agent.as_deref())
```

Comment:
the crlf / lf logic shall go inside of `CompletionService.generate`, instead of being implemented in `routes/…`